### PR TITLE
[UT] Add hive TPCH plan test

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [x] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. support mock Hive Table
2. add tpch plan test for hive 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Review by dirtysalt — DISMISSED** [review]

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by Youngwb** [comment]
run all

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 244 / 251 (97.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HiveResource.java|3|4|75.00%|[49]|
|:large_blue_circle:|com/starrocks/external/hive/MockHiveRepository.java|225|231|97.40%|[46, 47, 56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/external/hive/HiveColumnStats.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/physical/ScalarOperatorsReuse.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateResourceStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|

**Review by dirtysalt — DISMISSED** [review]

**Review by caneGuy — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/external/hive/MockHiveRepository.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.starrocks.sql.optimizer.Utils.getLongFromDateTime;
+import static java.lang.Double.NEGATIVE_INFINITY;
+import static java.lang.Double.POSITIVE_INFINITY;
+
+public class MockHiveRepository extends HiveRepository {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/MockHiveRepository.java:28]: You are my angel. but bad news bro, I will remove  the `HiveRepository` and `HiveMetaStoreTableInfo` in next version :(

**Review by stephen-shelby — DISMISSED** [review]

**Comment by Youngwb** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/external/hive/MockHiveRepository.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/MockHiveRepository.java:28]: 😭

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 244 / 251 (97.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HiveResource.java|3|4|75.00%|[49]|
|:large_blue_circle:|com/starrocks/external/hive/MockHiveRepository.java|225|231|97.40%|[46, 47, 56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/external/hive/HiveColumnStats.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/physical/ScalarOperatorsReuse.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateResourceStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/external/hive/MockHiveRepository.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/MockHiveRepository.java:28]: move to test dir

**Review by Seaven — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Pull request merged**